Serial

aidanahram opened the pull request:
changed the original result variable in the parseCommand function to a buffer as recommended by vsCode

Levi-Lesches reviewed (changes requested):
Forgot to mention the StringBuffer part earlier, hopefully that wasn't too bad to deal with. A few things: 
- Can you change your IDE to use tabs and re-commit `services.dart`?
- Make sure to add `Serial` to `services.dart`
- We probably should _not_ connect to serial in `init`, since the USB may not be in yet, or the user might want to change devices. Maybe put all that logic into another function, `connect`, which we can call whenever we want. That also means we'll need to add error handling to `sendBytes` and `incomingData` to throw some sort of error if `reader` and `writer` are still `null`. You can make your own errors by declaring a class that `implements Exception`. 

Other than that, all good and we should test this soon.

aidanahram commented:
Created a generic dashboardException class that can be inhereted by other exceptions to create an exception hierachy for the entire dashboard. Added a service to communicate over the serial port.

Changed Services.instance => services, and adds init to main
